src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -390,7 +404,16 @@ Payment::doApply()
             {
                 if (!view().exists(
                         keylet::depositPreauth(dstAccountID, account_)))
-                    return tecNO_PERMISSION;
+                {
+                    if (!credentialsPresent)
+                        return tecNO_PERMISSION;
+
+                    if (credentials::removeExpired(view(), ctx_.tx, j_))
+                        return tecEXPIRED;
+                    if (auto err = credentials::authorized(ctx_, dstAccountID);
+                        !isTesSuccess(err))
+                        return err;
+                }
             }
         }
 
```

Comment:
Note that MPT support needs to be added here.

Reply:
MPT added